shuttle/location.go

```diff
@@ -84,17 +84,14 @@ func (m *manager) GetLocationForStorage(ctx context.Context, obj cid.Cid, uid ui
 }
 
 func (cm *manager) primaryStagingLocation(ctx context.Context, uid uint) string {
-	var zones []util.Content
-	if err := cm.db.First(&zones, "user_id = ? and aggregate and not active", uid).Error; err != nil {
-		return ""
-	}
-
 	// TODO: maybe we could make this more complex, but for now, if we have a
 	// staging zone opened in a particular location, just keep using that one
-	for _, z := range zones {
-		return z.Location
+
+	var zone model.StagingZone
+	if err := cm.db.First(&zone, "user_id = ? and status = ?", uid, model.ZoneStatusOpen).Error; err != nil {
```

Comment:
this is nice, but not backward compatible, also, the zone is still being reconstructed. Best way would be to get the location of the most current content (desc). Also, we need to handle - not found

Reply:
i updated this to fallback to the old location logic when a StagingZone record is not found - does that address the backwards compatibility? i figure at some point we'll want to remove this fallback since it's relatively expensive and shouldn't be needed once the reconstruction process is done and those zones are aggregated

if there's no existing zone, i think it's correct to return empty string, since the caller will then pick the preferred shuttle from the shuttles table for which location to use next